### PR TITLE
Pandroid: add option to change screen gamepad, buttons position.

### PR DESCRIPTION
**Pull request by GabrielBRDeveloper**
# Changelog

- Now is possible create multiples screen gampead layout.
- Now is possible change screen gamepad buttons position and visibility. You can set a layout for portrait and landscape mode, only rotate screen or click in rotate screen button on editor.

# Screnshots

screenshot-2024-01-01_21 54 24 447

screenshot-2024-01-01_21 54 32 839

screenshot-2024-01-01_21 54 04 567

### CODE REVIEW

File: src/pandroid/app/src/main/java/com/panda3ds/pandroid/app/preferences/InputPreferences.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                        item.setSummary(profiles.get(which).getName());
+                    }).setTitle(R.string.pref_default_controller_title).show();
+        });
+
+        ((BaseActivity) requireActivity()).getSupportActionBar().setTitle(R.string.input);
+    }
+
+    public String formatName(String name) {
+        name = String.valueOf(name);
+        while (name.startsWith(" ")) {
```
Comment (wheremyfoodat) [inline src/pandroid/app/src/main/java/com/panda3ds/pandroid/app/preferences/InputPreferences.java:70]: Can't the 2 loops just be removed by using name.trim()

File: src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/controller/mapping/ControllerMapper.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    @Override
+    protected void dispatchDraw(Canvas canvas) {
+        super.dispatchDraw(canvas);
+        if (selectedView != null) {
+            drawSelected(canvas, selectionPaint);
+        }
+    }
+
+    public void drawSelected(Canvas canvas, Paint paint) {
+        int[] mePosition = new int[2];
```
Comment (wheremyfoodat) [inline src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/controller/mapping/ControllerMapper.java:90]: This naming is terrible

File: src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/controller/mapping/ControllerMapper.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        int[] viewPosition = new int[2];
+
+        selectedView.getLocationOnScreen(viewPosition);
+        getLocationOnScreen(mePosition);
+
+        int width = selectedView.getLayoutParams().width;
+        int height = selectedView.getLayoutParams().height;
+
+        int x = viewPosition[0] - mePosition[0];
```
Comment (wheremyfoodat) [inline src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/controller/mapping/ControllerMapper.java:99]: These are also dx and dy

Comment (wheremyfoodat) [reply src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/controller/mapping/ControllerMapper.java:99]: Actually no keep them x/y

**Comment by GabrielBRDeveloper** [comment]
Done
(https://github.com/wheremyfoodat/Panda3DS/pull/359/commits/1efb1491a611d9fea41085f202a59a0ddb01426d)

**Pull request merged**